packages/babel-parser/src/options.js

```diff
@@ -69,7 +70,13 @@ export const defaultOptions: Options = {
   // the parser sets extra.parenthesized on the expression nodes instead).
   createParenthesizedExpressions: false,
   // When enabled, errors are attached to the AST instead of being directly thrown.
-  // Some errors will still throw, because @babel/parser can't always recover.
+  // NOTE: it's not advisable to trust an AST when there are SyntaxErrors as some creative
+  // license was taken in order to recover from the error.
+  //
+  // When `false` errors are thrown.
+  // When `true` some errors will still throw, because @babel/parser can't always recover.
+  // When `"always"` errors will never throw and an AST will always be returns with errors
+  // attached to the AST.
```

Comment:
```suggestion
  // When `true` @babel/parser tries to recover from errors, but throws when it can't.
  // When `"always"` @babel/parser never throws, but it returns an empty AST node when it can't recover from parsing errors.
```